syntax = "proto3";


message ProtoContact
{
    int32 proto_prime = 1;
}

message ProtoHello
{
    string proto_message = 1;
}

message ItemHeader
{
    int32 proto_id = 1;
    int32 proto_type = 2;
    int32 proto_size = 3;
}

message ProtoHeader
{
    repeated ItemHeader protoItems = 1;
}

message ProtoMapping
{
    int32 key = 1;
    string value = 2;
}

message ProtoJoin
{
    repeated ProtoMapping ixToProtos = 1;
}

message Wall
{
    bool visible = 1;
    ProtoVector2 p1 = 2;
    ProtoVector2 p2 = 3;
}

message ProtoVector2
{
    float x = 1;
    float y = 2;
}

message BuildMazeMsg
{
    string maze_name = 1;
    ProtoVector2 start_point = 2;
    ProtoVector2 goal_point = 4;
    repeated Wall walls = 5;
}
